### PR TITLE
Update SmokaResolver to be SubaruResolver...

### PR DESCRIPTION
**Pull request by hjeeves**
... plus changes to the 'file' URI, so it's now more reflective of handling raw data.

### CODE REVIEW

File: caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolver.java [hunk 1]
```diff
@@ -40,27 +41,18 @@
...
-        + "=&date_obs=&exptime=&observer=&prop_id=&frameid=&dataset=&asciitable=Table"
-        + "&frameorshot=Frame&action=Search&instruments=SUP&instruments=HSC&multiselect_0=SUP&multiselect_0=HSC"
-        + "&multiselect_0=SUP&multiselect_0=HSC&obs_mod=IMAG&obs_mod=SPEC&obs_mod=IPOL&multiselect_1=IMAG&multiselect_1=SPEC"
-        + "&multiselect_1=IPOL&multiselect_1=IMAG&multiselect_1=SPEC&multiselect_1=IPOL&data_typ=OBJECT&multiselect_2=OBJECT"
-        + "&multiselect_2=OBJECT&bandwidth_type=FILTER&band=&dispcol=FRAMEID&dispcol=DATE_OBS&dispcol=FITS_SIZE&dispcol=OBS_MODE"
-        + "&dispcol=DATA_TYPE&dispcol=OBJECT&dispcol=FILTER&dispcol=WVLEN&dispcol=DISPERSER&dispcol=RA2000&dispcol=DEC2000"
-        + "&dispcol=UT_START&dispcol=EXPTIME&dispcol=OBSERVER&dispcol=EXP_ID&orderby=FRAMEID&diff=100&output_equinox=J2000&from=0"
-        + "&exp_id="; //&exp_id=SUPE01318470 or similar for last entry here.
+
+    private static final String BASE_DATA_URL = "http://www.canfar.net/maq/subaru?frameinfo=";
```
Comment (pdowler) [inline caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolver.java]: The whole MAQ project is a CADC activity. Why is this page hosted on www.canfar.net?

File: caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolver.java [hunk 2]
```diff
@@ -88,20 +80,21 @@ public URL toURL(URI uri) {
...
+        if (path.length == 2 && requestType.equals(PREVIEW_URI)) {
             // Returns a web page reference
-            sb = PREVIEW_BASE_URL + "?" + PREVIEW_URL_QUERY + fileName;
+            sb = PREVIEW_BASE_URL + "?" + PREVIEW_URL_QUERY + path[1];
+        } else if (path.length == 3 && requestType.equals(RAW_DATA_URI)) {
+            // expected URI input is subaru:data/YYYY-MM-dd/<FRAMEID>
+            // expected URL output is http://www.canfar.net/maq/subaru?frameinfo=YYYY-MM-dd%20FRAMEID
+            sb = BASE_DATA_URL +  NetUtil.encode(path[1] + "/" + path[2]);
         } else {
             throw new IllegalArgumentException("Invalid URI. Expected 'file' or 'preview' and got " + requestType);
```
Comment (pdowler) [inline caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolver.java]: The mention of 'file' in the error message doesn't match declared constant values; that will be misleading if (when) it happens

File: caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolver.java [hunk 3]
```diff
@@ -88,20 +80,21 @@ public URL toURL(URI uri) {
...
-            // Returns a quick search-style URL for SMOKA Search page
-            sb = FILE_BASE_URL + "?" + FILE_URL_QUERY + fileName;
-        } else if (requestType.equals(PREVIEW_URI)) {
+        String requestType = path[0];
+        if (path.length == 2 && requestType.equals(PREVIEW_URI)) {
             // Returns a web page reference
-            sb = PREVIEW_BASE_URL + "?" + PREVIEW_URL_QUERY + fileName;
+            sb = PREVIEW_BASE_URL + "?" + PREVIEW_URL_QUERY + path[1];
+        } else if (path.length == 3 && requestType.equals(RAW_DATA_URI)) {
+            // expected URI input is subaru:data/YYYY-MM-dd/<FRAMEID>
```
Comment (pdowler) [inline caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolver.java]: this comment has a path value that doesn't match declared constants

File: caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolver.java (hunk 1 above)
Comment (at88mph) [reply caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/SubaruResolver.java]: It was easier to get up and running.  Is it misleading?

**Pull request merged**